frontend/src/lib/components/Users/CreateUser.svelte

```diff
@@ -0,0 +1,98 @@
+<script lang="ts">
+  import PasswordStrengthMeter from '$lib/components/PasswordStrengthMeter.svelte';
+  import { SubmitButton, FormError, Input, Form, ProtectedForm, isEmail, lexSuperForm, passwordFormRules, DisplayLanguageSelect } from '$lib/forms';
+  import t, { getLanguageCodeFromNavigator, locale } from '$lib/i18n';
+  import { type RegisterResponse } from '$lib/user';
+  import { getSearchParamValues } from '$lib/util/query-params';
+  import { createEventDispatcher, onMount } from 'svelte';
+  import { usernameRe } from '$lib/user';
+  import { z } from 'zod';
+
+  export let allowUsernames = false;
+  export let skipTurnstile = false;
+  export let submitButtonText = $t('register.button_register');
+  export let handleSubmit: (password: string, passwordStrength: number, name: string, email: string, locale: string, turnstileToken: string) => Promise<RegisterResponse>;
+
+  const dispatch = createEventDispatcher();
+
+  type RegisterPageQueryParams = {
+    name: string;
+    email: string;
+  };
+  let turnstileToken = '';
+  // $locale is the locale that our i18n is using for them (i.e. the best available option we have for them)
+  // getLanguageCodeFromNavigator() gives us the language/locale they probably actually want. Maybe we'll support it in the future.
+  const userLocale = getLanguageCodeFromNavigator() ?? $locale;
+  const formSchema = z.object({
+    name: z.string().trim().min(1, $t('register.name_missing')),
+    email: z.string().trim()
+      .min(1, $t('project_page.add_user.empty_user_field'))
+      .refine((value) => isEmail(value) || (allowUsernames && usernameRe.test(value)), { message: $t('register.invalid_input') }),
```

Comment:
I think it's important to tell the user what the username rules are. And the username validation warning should be seperate from the email warning. We already have a common email warning (`form.invalid_email`).

We already have a message for the bulk tool that includes the rules. For this I'd reword it to: `Invalid login/username. Only letters, numbers, and underscore (_) characters are allowed.`

Note: I've changed the wording of the second sentence. Please use either this wording or the current wording for both. Pick whichever you like most.

Reply:
Done in commit f027c6bc. Is that what you had in mind?

Reply:
It was a bit buggy, so I added [f2d011a](https://github.com/sillsdev/languageforge-lexbox/pull/736/commits/f2d011a30b27f1d918f3b7bbbd3d994727c6cde6).